### DISCUSSION TITLE
Binaries in /target/release

### DISCUSSION THREAD

**Discussion by 0323pin** [ast-grep/ast-grep#847] [General]
Hi,

I would like to finish the NetBSD package for `ast-grep` that has been in our wip-repository for a few months.

It builds fine and, I've tested the `sg` command after updating the package to v0.17.0 today.
Now, I wonder should we only install the `sg` binary (the only one referenced in the README.md) or, should we also install `ast-grep`? I can see that the build produces two binaries in `/target/release`.

Thank you for any hints.

**Comment by HerringtonDarkholme** [comment]
Hi @0323pin, thank you for packaging ast-grep! I really appreciate it!

I would recommend install both `sg` and `ast-grep`, as discussed in https://github.com/ast-grep/ast-grep/issues/56.

`sg` was taken by Linux utils decades ago so it is better to ship two commands. I recommend shipping both for non-Linux OS as well to avoid confusion.

Thanks!

**Comment by 0323pin** [comment]
@HerringtonDarkholme Thanks! I'll change the package to install both binaries. Also, thanks for linking #56, I'll grep the package tree to be sure if, there are conflicting binaries in tree. If so, any preferences for a _re-name_ during installation?

**Reply by HerringtonDarkholme** [reply]
If that was the case, would you like to ship only `ast-grep`? Thanks!

**Reply by 0323pin** [reply]
I've just did grep the package tree and there are no issues. So, I'll change the package to install both binaries. I'll try to get this done during the weekend and merge the package into the main tree.

Thanks.

**Comment by 0323pin** [comment]
@HerringtonDarkholme I've just updated the package to 0.17.1 but, I still have one question if you don't mind.

`cmp` tells me that the two binaries are not the same. Also, they have different checksums. Would you mind sharing some light on what they differ? I would prefer not to try to diff hexabits.

Thanks

**Reply by HerringtonDarkholme** [reply]
I don't have any idea why they are different. They are located in different directory so maybe cargo will build them differently, say symbol info. 

You can try `vimdiff -y <(xxd -l2000 ast-grep) <(xxd -l2000 sg)` , I saw many diffs though

**Reply by 0323pin** [reply]
Thanks!

**Comment by 0323pin** [comment]
@HerringtonDarkholme [Merged into main](http://mail-index.netbsd.org/pkgsrc-changes/2024/01/13/msg290784.html)
